contributors/design-proposals/componentconfig-dump.md

```diff
@@ -0,0 +1,164 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should refer to the docs that go with that version.
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+We describe a proposal for retrieving component related configurations (componentconfigs and otherwise) that govern the behavior of any one of the Kubernetes (K8s) components.
+
+We propose several important changes:
+
+* adding new authenticated protected path to a component api:
+    * `/configs`
```

Comment:
I think that's a larger discussion we should have. ConfigMap in its current form might be too low-level - you basically need to pass blobs around to have nested data and if you want schema, schema versioning, you need to roll your own enforcement. Structured, typed data seems more useful, natural, and maintainable. We have mechanisms for dealing with structured data for core components. For non-core ThirdPartyResource may be a solution. Random thought - why (beyond historical reasons) isn't everything built as a "third party resource" (and likely namespaced to separate core from non-core)?

I'd also be interested in seeing some data on how people are using ConfigMap in production today before we make a decision about its longevity.